octodns/provider/cloudflare.py

```diff
@@ -65,8 +77,11 @@ def _request(self, method, path, params=None, data=None):
         resp = self._sess.request(method, url, params=params, json=data,
                                   timeout=self.TIMEOUT)
         self.log.debug('_request:   status=%d', resp.status_code)
+        if resp.status_code == 400:
+            raise CloudflareError(resp.json())
```

Comment:
There doesn't appear to be any test cases running into this 400 case. Should be relatively simple to get one in place and then this will be good to go.